docs/source/1.0/spec/core/behavior-traits.rst

```diff
@@ -655,3 +655,228 @@ See
+
+        .. code-block:: abnf
+
+            prefix = lower-alpha *(["-"](lower-alpha / DIGIT)) "-"
+            lower-alpha = %x61-7A
+
+        That is, it should start with a lowercase letter followed by any
+        number of lowercase letters, digits, or non-sequential hyphens,
+        and it should end in a hyphen.
+
+        The prefix string MUST NOT match the prefix value used with the
+        ``httpPrefixHeaders`` trait when modeled within the same HTTP request
+        or HTTP response enclosure. Additionally, any HTTP header binding
+        within the same HTTP request or HTTP response enclosure SHOULD NOT
+        start with the prefix string. This is to avoid any unintentional data
+        overrides.
+
+    * - locations
+      - ``[string]``
+      - A priority-ordered list representing supported locations where checksum
+        can be supplied. Valid values are:
+
+        * ``header`` - Indicates the checksum should be placed in an HTTP header.
+        * ``trailer`` - Indicates the checksum should be placed in the `chunked trailer part`_
+          of the body.
+
+        The default value is a list only containing ``header``.
+
+    * - algorithms
+      - ``[string]``
+      - **Required**. List of string representing checksum algorithms
+        supported for the HTTP request or response. A valid algorithm
+        should follow ABNF:
+
+        .. code-block:: abnf
+
+            algorithms = 1*(lower-alpha / DIGIT)
+            lower-alpha = %x61-7A
+
+        That is, it should be comprised only of digits and lowercase letters.
+
+.. _chunked trailer part: https://tools.ietf.org/html/rfc7230#section-4.1.2
+
+
+.. _httpChecksum_trait_behavior:
+
+HttpChecksum Behavior
+=====================
+
+.. rubric:: Client Behavior
+
+If the ``httpChecksum`` trait has a modeled ``request`` section, for HTTP request,
+the client MUST compute the request payload checksum, using an algorithm
+defined in the algorithms property. This computed checksum MUST be supplied
+at one of the locations defined in the location property of the trait. It SHOULD be
+supplied at the first supported location defined in the locations property of the trait.
+If the resolved location is ``header``, the client MUST put the checksum into the
+HTTP request headers. If the resolved location is ``trailer``, the client MUST put
+the checksum into the `chunked trailer part`_ of the body. The header or trailer
+field name MUST be constructed using the prefix property as described in the
+:ref:`HTTP checksum properties<checksum-properties>` section.
+
+If the ``httpChecksum`` trait has a modeled ``response`` section, for HTTP
+response, the client MUST look for a checksum at supported locations as per
+defined properties. If a checksum is found, the client MUST validate the received
+checksum value by computing the corresponding checksum of the received payload.
+
+Service MAY model input or output shape member bound to an HTTP header
```

Comment:
An operation's `input`, `output`, and `error` structures MAY contain a member bound...